packages/catalog-client/src/generated/apis/DefaultApi.client.ts

```diff
@@ -0,0 +1,828 @@
+    }
+    if (isCodeInRange('201', response.status)) {
+      return await response.json();
+    }
+    if (isCodeInRange('400', response.status)) {
+      return (await response.text()) as any;
+    }
+    if (isCodeInRange('0', response.status)) {
+      return (await response.text()) as any;
+    }
+    return await response.json();
+  }
+
+  /**
+   * Delete a single entity by UID.
+   * @param uid
+   */
+  public async deleteEntityByUid(
+    uid: string,
+    options?: RequestOptions,
+  ): Promise<void> {
+    const baseUrl = await this.discoveryApi.getBaseUrl(pluginId);
+    const headers: Record<string, any> = {};
+    const contentType = 'application/json';
+    const body = null;
+
+    const uriTemplate = `/entities/by-uid/{uid}`;
+    const template = parser.parse(uriTemplate.replaceAll(' ', ''));
+
+    const uri = template.expand({
+      uid: uid,
+    });
+    headers.Accept = 'application/json, */*;q=0.8';
+
+    const url = `${baseUrl}${uri}`;
+    const response = await this.fetchApi.fetch(url, {
+      headers: {
+        ...headers,
+        'Content-Type': contentType,
+        ...(options?.token && { Authorization: `Bearer ${options?.token}` }),
+      },
+      method: 'DELETE',
+      body,
+    });
+    if (!response.ok) {
+      throw new ResponseError(response);
+    }
+    if (isCodeInRange('204', response.status)) {
+      return (await response.text()) as any;
+    }
+    if (isCodeInRange('400', response.status)) {
+      return (await response.text()) as any;
+    }
+    if (isCodeInRange('0', response.status)) {
+      return (await response.text()) as any;
+    }
+    return await response.json();
+  }
+
+  /**
+   * Delete a location by id.
+   * @param id
+   */
+  public async deleteLocation(
+    id: string,
+    options?: RequestOptions,
+  ): Promise<void> {
+    const baseUrl = await this.discoveryApi.getBaseUrl(pluginId);
+    const headers: Record<string, any> = {};
+    const contentType = 'application/json';
+    const body = null;
+
+    const uriTemplate = `/locations/{id}`;
+    const template = parser.parse(uriTemplate.replaceAll(' ', ''));
+
+    const uri = template.expand({
+      id: id,
+    });
+    headers.Accept = 'application/json, */*;q=0.8';
+
+    const url = `${baseUrl}${uri}`;
+    const response = await this.fetchApi.fetch(url, {
+      headers: {
+        ...headers,
+        'Content-Type': contentType,
+        ...(options?.token && { Authorization: `Bearer ${options?.token}` }),
+      },
+      method: 'DELETE',
+      body,
+    });
+    if (!response.ok) {
+      throw new ResponseError(response);
+    }
+    if (isCodeInRange('204', response.status)) {
+      return (await response.text()) as any;
+    }
+    if (isCodeInRange('400', response.status)) {
+      return (await response.text()) as any;
+    }
+    if (isCodeInRange('0', response.status)) {
+      return (await response.text()) as any;
+    }
+    return await response.json();
```

Comment:
Just a note that I think we could think a about exactly what to return here. I'm feeling that a `ResponseError.fromResponse(res)` *should* make sense here but maybe it doesn't. Also feeling we likely want to return the response object as is rather than the body. There may be situations we want to access headers and whatnot.

Reply:
I like the idea of switching to `ResponseError.fromResponse(res)`. I remembered the issue I was having though, basically, `validate-entity` has a weird response shape of `{errors : Error[]}` that doesn't match the generic error response. Open to suggestions here, my thinking is either to add a `rawBody` property to `ResponseError` (probably better) and then parse from there or adjust the `validate-entity` return value (less good)